src/coordinator/coordinator.hpp

```diff
@@ -0,0 +1,270 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <unordered_set>
```

Comment:
```suggestion
#include <optional>
#include <string>
#include <unordered_set>
#include <variant>
#include <vector>
```
`<variant>` for `std::visit`